mlflow/tracking/fluent.py

```diff
@@ -393,6 +449,11 @@ def start_run(
     elif run_id := MLFLOW_RUN_ID.get():
         existing_run_id = run_id
         del os.environ[MLFLOW_RUN_ID.name]
+    # Get SGC job run ID tag key for run resumption if applicable
+    elif sgc_job_run_id_tag_key := _get_sgc_job_run_id_tag_key():
```

Comment:
didn't notice sgc_job_run_id_tag_key is used in line 547. Let's add `sgc_job_run_id_tag_key: str | None = None` to avoid a not-defined error.